api/src/main/java/au/org/aodn/nrmn/restapi/controller/IngestionController.java

```diff
@@ -25,84 +26,157 @@
+    protected MaterializedViewService materializedViewService;
 
     @Autowired
-    private MaterializedViewService materializedViewService;
+    protected GlobalLockService globalLockService;
 
     @Autowired
-    private GlobalLockService globalLockService;
+    private PlatformTransactionManager transactionManager;
 
-    private static Logger logger = LoggerFactory.getLogger(IngestionController.class);
+    protected  TransactionTemplate tx;
+
+    protected ExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+
+    @PostConstruct
+    public void initialization() {
+        tx = new TransactionTemplate(transactionManager);
+    }
+
+    @PreDestroy
+    public void closeDown() {
+        executorService.shutdown();
+    }
 
     @PostMapping(path = "ingest/{job_id}")
     @Operation(security = { @SecurityRequirement(name = "bearer-key") })
-    public ResponseEntity<String> ingest(@PathVariable("job_id") Long jobId) {
+    public ResponseEntity<Map> ingest(@PathVariable("job_id") Long jobId) {
         userActionAuditRepository.save(new UserActionAudit("ingestion/ingest", "ingest job: " + jobId));
 
-        var optionalJob = jobRepository.findById(jobId);
-        if (!optionalJob.isPresent())
-            return ResponseEntity.badRequest().body("Job with given id does not exist. jobId: " + jobId);
-
-        var job = optionalJob.get();
-        if (job.getStatus() != StatusJobType.STAGED)
-            return ResponseEntity.badRequest().body("Job with given id has not been validated: " + jobId);
-
-        if (!globalLockService.setLock())
-            return ResponseEntity.ok("locked");
+        Map<String, Object> result = new HashMap<>();
+        result.put("jobId", jobId);
 
-        try {
+        var optionalJob = jobRepository.findById(jobId);
 
-            var stagedJobLog = StagedJobLog.builder()
-                    .stagedJob(job)
-                    .eventType(StagedJobEventType.INGESTING)
-                    .build();
-            stagedJobLogRepository.save(stagedJobLog);
+        if (!optionalJob.isPresent()) {
+            result.put("jobStatus", StatusJobType.FAILED);
+            result.put("message", "Job with given id does not exist. jobId: " + jobId);
+            logger.warn(result.get("message").toString());
+            return ResponseEntity.badRequest().body(result);
+        }
 
-            var rows = rowRepository.findRowsByJobId(job.getId());
-            var species = speciesFormatting.getSpeciesForRows(rows);
-            var validatedRows = speciesFormatting.formatRowsWithSpecies(rows, species);
+        final var job = optionalJob.get();
+        if (job.getStatus() != StatusJobType.STAGED) {
+            result.put("jobStatus", StatusJobType.FAILED);
+            result.put("message", "Job with given id has not been validated: " + jobId);
+            logger.warn(result.get("message").toString());
+            return ResponseEntity.badRequest().body(result);
+        }
```

Comment:
Can we also reject the request if a job has been validated but with `BLOCKING` validation errors?